File: my_first_py/my_first_py/my_first_subscriber.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
class RobotNewsSubscriberNode(Node):
    def __init__(self):
        super().__init__("robot_news_subscriber")
        self.subscriber_= self.create_subscription(String,"robot_news_msg",self.callback_robot_news_msg,10)
        self.counter_=0
        self.get_logger().info("Robot Has Started Reciving Msg:")
    def callback_robot_news_msg(self,msg: String):
        self.get_logger().info(f"{self.counter_}"+"."+msg.data)
        self.counter_+=1
    
def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsSubscriberNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ =="__main__":
    main()

File: my_first_cpp/src/my_first_subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsSubscriberNode : public rclcpp::Node 
{
public:
    RobotNewsSubscriberNode() : Node("robot_news_subscriber") {
        subscriber_ =  this->create_subscription<example_interfaces::msg::String>("robot_news_msg",10,
        std::bind(&RobotNewsSubscriberNode::callback_robot_news,this,std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(),"Robot News Station Is Listening:");
    }
private:
    void callback_robot_news(const example_interfaces::msg::String::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(),"%s",msg->data.c_str());
    }
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsSubscriberNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}